avm/src/daemon/daemon.rs

```diff
@@ -26,29 +29,33 @@ use crate::vm::run::run;
 pub static CLUSTER_SECRET: OnceCell<Option<String>> = OnceCell::new();
 
 #[cfg(target_os = "linux")]
-async fn get_private_ip() -> String {
-  let res = Command::new("hostname").arg("-I").output().await;
-  let err = "Failed to execute `hostname`";
-  let stdout = res.expect(err).stdout;
-  String::from_utf8(stdout)
-    .expect(err)
-    .trim()
-    .split_whitespace()
-    .next()
-    .unwrap()
-    .to_string()
+async fn get_private_ip() -> Result<String, Box<dyn Error + Send + Sync>> {
+  let res = Command::new("hostname").arg("-I").output().await?;
+  let stdout = res.stdout;
+  let private_ip = String::from_utf8(stdout);
+  match private_ip {
+    Ok(private_ip) => match private_ip.trim().split_whitespace().next() {
+      Some(private_ip) => Ok(private_ip.to_string()),
+      None => return Err("No ip found".into()),
+    },
+    Err(err) => return Err(err.into()),
```

Comment:
nit: you have 2 `return`s here even though all 3 arms return something. Should be consistent between arms.